### PR TITLE
Unify consensus subcommands

### PR DESCRIPTION
**Pull request by FelixMoelder**
This PR includes the following changes:

- Merged call-consensus-reads and call-overlapping-consensus-reads subcommands into a single one.
- Implemented two additional writers for calling overlapping consensus reads. Consensuses of read pairs with no overlaps are written into these.
- Fixed a bug where the relative instead of absolute hamming distance was calculated. This resulted in bypassing the filtering step leading to increased runtimes.

### CODE REVIEW

File: src/fastq/call_consensus_reads/pipeline.rs [hunk 1]
```diff
@@ -398,6 +411,37 @@ impl<'a, R: io::Read, W: io::Write> CallOverlappingConsensusRead<'a, R, W> {
             reverse_umi,
         }
     }
+    pub fn likeliest_nonoverlapping_consensus(
```
Comment (johanneskoester) [inline src/fastq/call_consensus_reads/pipeline.rs]: maximum_likelihood_nonoverlapping_consensus

File: src/fastq/call_consensus_reads/pipeline.rs [hunk 2]
```diff
@@ -419,36 +463,46 @@ impl<'a, R: io::Read, W: io::Write> CallConsensusReads<'a, R, W>
...
-                        .calc_consensus(),
+                    let (consensus_record, lh) = CalcOverlappingConsensus::new(
+                        &f_recs,
+                        &r_recs,
+                        overlap,
+                        &outer_seqids,
+                        &uuid,
                     )
+                    .calc_consensus();
+                    Some((consensus_record, lh, *insert_size))
```
Comment (johanneskoester) [inline src/fastq/call_consensus_reads/pipeline.rs]: Return a struct here. Put it in the variable `consensus_overlap`. Return a struct below, put it in the variable `consensus_no_overlap`. THe struct should be like this:
```
pub struct Consensus {
    record: ...,
    likelihood: ...,
    insert_size: ...
}
```

Thereby likelihood is the likelihood of the consensus sequence, i.e., Pr(R, I | C) = Pr(R | C, I) * Pr(I).

**Comment by FelixMoelder** [comment]
The write_records function should be more readable now, but there are still a few things left to do.
I'll report back when I'm done cleaning up.

**Comment by FelixMoelder** [comment]
The naming of variables should be now more intuitive.
Both maximum_likelihood functions now follow a more similar pattern improving readability and comparability.
Are there any more remarks?

File: src/fastq/call_consensus_reads/pipeline.rs [hunk 3]
```diff
@@ -100,6 +106,33 @@ impl FASTQStorage {
...
+}
+
+pub struct NonOverlappingConsensus {
+    f_record: Record,
+    r_record: Record,
+    likelihood: LogProb,
+}
+
+impl NonOverlappingConsensus {
+    pub fn new(f_record: Record, r_record: Record, likelihood: LogProb) -> Self {
```
Comment (johanneskoester) [inline src/fastq/call_consensus_reads/pipeline.rs]: You can remove the constructors, and directly initialize the structs. We anyway use them only internally.

**Pull request merged**